docs/source/integration/integration.rst

```diff
@@ -0,0 +1,261 @@
+
+
+
+Writing your own :code:`Extractor` and :code:`GraphNeTFileReader`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+    We recommend visiting the :code:`DataConverter` documentation before reading this example
+
+Suppose you want to add a reader and extractor for MyExperiment and that the data
+is stored as pickled dictionaries with two kinds of entries:
+
+**hits**: a single pandas.DataFrame with dimensions [n_pulses,d] where `n_pulses` is the number of observed Cherenkov pulses across all events in the file, and `d` is the number of features we know about each measurement.
+These features would normally include the sensor position, time of measurement, event id etc. 
+
+**truth**: a single pandas.DataFrame with dimensions [n_events, t] where `n_events` denotes the total number of events in the file and `t` is the number of event-level truth variables we have available for this simulation.
+Ordinarily, these truth variables would include particle id, energy, direction etc. 
+
+
+To convert these pickle files to a supported backend in GraphNeT, we first have to define our reader. This reader should open a pickle file and apply the :code:`Extractor` that we must also implement. Lets start with the reader:
+
+.. code-block:: python
+
+    from typing import List, Union, Dict
+    import pandas as pd
+    import pickle
+
+    # Import the generic file reader
+    from .graphnet_file_reader import GraphNeTFileReader
+
+    # Import your own extractor
+    from graphnet.data.extractors.myexperiment import MyExtractor
+
+    class MyReader(GraphNeTFileReader):
+        """A class for reading my pickle files from MyExperiment."""
+
+        _accepted_file_extensions = [".pickle"]
+        _accepted_extractors = [MyExperiment]
+
+        def __call__(self, file_path: str) -> Dict[str, pd.DataFrame]:
+            """Extract data from single pickle file.
+
+            Args:
+                file_path: Path to pickle file.
+
+            Returns:
+                Extracted data.
+            """
+
+            # Open file
+            file = open(file_path,'r')
+            data = pickle.load(file)
+
+            # Apply extractors
+            outputs = {}
+            for extractor in self._extractors:
+                output = extractor(data)
+                if output is not None:
+                    outputs[extractor._extractor_name] = output
+            return outputs
+
+When the :code:`DataConverter` is instantiated, it will set the :code:`Extractors` that it was instantiated with as member
+variables of our :code:`GraphNeTFileReader`, making them available to us under :code:`self._extractors`. When the conversion is running, 
+the :code:`DataConverter` will pass a :code:`file_path` to our :code:`__call__` function, and it is the job of our reader to open this
+file and apply extractors to it. These calls will happen in parallel automatically, so we do not have to worry about parallelizing this code ourselves.
```

Comment:
Probably just finish the sentence after "These calls will happen in parallel automatically". The final point is relatively obvious.